features/telemetry/listeners/sflow/src/main/java/org/opennms/netmgt/telemetry/listeners/sflow/Listener.java

```diff
@@ -0,0 +1,135 @@
+/*******************************************************************************
+ * This file is part of OpenNMS(R).
+ *
+ * Copyright (C) 2017-2017 The OpenNMS Group, Inc.
+ * OpenNMS(R) is Copyright (C) 1999-2017 The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is a registered trademark of The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License,
+ * or (at your option) any later version.
+ *
+ * OpenNMS(R) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with OpenNMS(R).  If not, see:
+ *      http://www.gnu.org/licenses/
+ *
+ * For more information contact:
+ *     OpenNMS(R) Licensing <license@opennms.org>
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+package org.opennms.netmgt.telemetry.listeners.sflow;
+
+import java.net.InetSocketAddress;
+
+import org.opennms.core.ipc.sink.api.AsyncDispatcher;
+import org.opennms.netmgt.telemetry.listeners.api.TelemetryMessage;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.netty.bootstrap.Bootstrap;
+import io.netty.channel.ChannelFuture;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.ChannelInboundHandlerAdapter;
+import io.netty.channel.ChannelInitializer;
+import io.netty.channel.ChannelOption;
+import io.netty.channel.EventLoopGroup;
+import io.netty.channel.nio.NioEventLoopGroup;
+import io.netty.channel.socket.DatagramChannel;
+import io.netty.channel.socket.nio.NioDatagramChannel;
+import io.netty.util.internal.SocketUtils;
+
+public class Listener implements org.opennms.netmgt.telemetry.listeners.api.Listener {
+
+    private static final Logger LOG = LoggerFactory.getLogger(Listener.class);
+
+    private String name;
+
+    private String host = null;
+    private int port = 6343;
+
+    private AsyncDispatcher<TelemetryMessage> dispatcher;
+
+    private EventLoopGroup bossGroup;
+    private ChannelFuture socketFuture;
+
+    public Listener() {
+    }
+
+    public void start() throws InterruptedException {
+        this.bossGroup = new NioEventLoopGroup();
+
+        final InetSocketAddress address = this.host != null
+                                ? SocketUtils.socketAddress(this.host, this.port)
+                                : new InetSocketAddress(this.port);
+
+        this.socketFuture = new Bootstrap()
+                .group(this.bossGroup)
+                .channel(NioDatagramChannel.class)
+                .option(ChannelOption.SO_REUSEADDR, true)
```

Comment:
Can we setup the listener using the same options as the `org.opennms.netmgt.telemetry.listeners.udp.UdpListener`? (We should probably update `org.opennms.netmgt.telemetry.listeners.flow.AbstractUdpListener` with those as well.)